src/MemoryRouter.tsx

```diff
@@ -175,10 +175,12 @@ export class MemoryRouter extends BaseRouter {
     // Update this instance:
     this.asPath = asPath;
     if (asRoute) {
+      this.route = asRoute.pathname;
       this.pathname = asRoute.pathname;
       this.query = rawPathnamesDiffer ? asRoute.query : newRoute.query;
       this.hash = asRoute.hash;
     } else {
+      this.route = newRoute.pathname;
```

Comment:
Since there's no real logic to `route`, I think it'd be better to implement as a simple `get route() { return this.pathname; }` to make it clear that it's always the same.  What do you think?

Reply:
I've updated the implementation, let me know if it makes sense.